### PR TITLE
chore: adding git hooks

### PR DESCRIPTION
**Pull request by ISHIMWE-Aime**
# Description
I have added pre-commit and pre-push git hooks.
also I added git action (CI) to run tests

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm test
```
Comment (mucyoalexaxel) [inline .husky/pre-commit:4]:
```suggestion
pnpm test
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@emilenfc check the suggested changes.

**Review by emilenfc — DISMISSED** [review]
LGTM

**Review by emilenfc — APPROVED** [review]
LGTM

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**